app/controllers/sign_up/email_resend_controller.rb

```diff
@@ -1,22 +1,36 @@
 module SignUp
   class EmailResendController < ApplicationController
-    include ValidEmailParameter
     include UnconfirmedUserConcern
 
     def new
       @user = User.new
+      @resend_email_confirmation_form = ResendEmailConfirmationForm.new('')
     end
 
     def create
-      User.send_confirmation_instructions(user_params)
-      flash[:notice] = t('devise.confirmations.send_paranoid_instructions')
-      redirect_to root_path
+      @resend_email_confirmation_form = ResendEmailConfirmationForm.new(downcased_email)
+      result = @resend_email_confirmation_form.submit
+
+      analytics.track_event(Analytics::RESEND_EMAIL_CONFIRMATION, result)
+
+      if result[:success]
+        handle_valid_email
+      else
+        render :new
+      end
     end
 
     private
 
-    def user_params
-      params.fetch(:user, {})
+    def downcased_email
+      params[:resend_email_confirmation_form][:email].downcase
```

Comment:
I don't think so. The tests will fail, which is what we want. The only way a user would get an error is if they change the HTML before submitting the form.